### ISSUE TITLE
Silent failure when specifying multiple directories with `--no-error-on-unmatched-pattern`

### ISSUE THREAD

**Issue by davidtaylorhq** [ember-template-lint/ember-template-lint#2758]
When invoking `ember-template-lint` like this:

```
ember-template-lint --no-error-on-unmatched-pattern dirA dirB
```

I would expect it to lint any files in `dirA` or `dirB`. If one of the directories doesn't have any `.hbs` files, I would still expect it to lint files in the other directory.

In reality, if either one of the directories doesn't have `hbs` files, the command will fail silently.

Relevant exception:

https://github.com/ember-template-lint/ember-template-lint/blob/8a11c93b01093fd729390e94eb743cbc58d74c12/lib/helpers/cli.js#L194

Which is caught here:

https://github.com/ember-template-lint/ember-template-lint/blob/8a11c93b01093fd729390e94eb743cbc58d74c12/bin/ember-template-lint.js#L178-L179
